index.html

```diff
@@ -9382,12 +9411,165 @@
+              to the network. This includes any data sent during the execution of the
+              current task, regardless of whether the <a>user agent</a> is able to
+              transmit text asynchronously with script execution. This does not
+              include framing overhead incurred by the protocol, or buffering done
+              by the operating system or network hardware. On getting, it
+              <em class="rfc2119" title="MUST">MUST</em> return the value of the
+              <code><a>RTCQuicStream</a></code>'s <a>[[\WriteBufferedAmount]]</a> internal slot.
+              If the <code><a>RTCQuicStream</a></code> is in the <code>closed</code>
+              state, this attribute's value will only increase with each call to the
+              <code>write</code> method (the attribute does not reset to zero once the
+              <code><a>RTCQuicStream</a></code> closes).</p>
+            </dd>
+            <dt><code>maxWriteBufferedAmount</code> of type <span class="idlAttrType"><a>unsigned
+            long</a></span>, readonly</dt>
+            <dd>
+              <p>The <dfn id="dom-quicstream-maxwritebufferedamount"><code>maxWriteBufferedAmount</code></dfn>
+              attribute represents the maximum number of bytes that the implementation allows
+              to be buffered by <code>write</code>. On getting, it <em class="rfc2119" title="MUST">MUST</em> 
+              return the value of the <code><a>RTCQuicStream</a></code>'s
+              <a>[[\MaxWriteBufferedAmount]]</a> internal slot.</p>
+            </dd>
           </dl>
        </section>
        <section>
           <h2>Methods</h2>
           <dl data-link-for="RTCQuicStream" data-dfn-for="RTCQuicStream" class=
           "methods">
+            <dt><dfn><code>readInto</code></dfn></dt>
+            <dd>
+              <p>Reads from <code><a>RTCQuicStream</a></code> into the buffer specified
+              by the first argument and returns the number of bytes read.
+              When the <code>readInto</code> method is called,
+              the user agent <em class="rfc2119" title="MUST">MUST</em> run the following
+              steps:</p>
+              <ol>
+               <li>Let <var>stream</var> be the <code><a>RTCQuicStream</a></code> object
+               on which <code>readInto</code> is invoked.</li>
+               <li>If <var>stream</var>'s <a>[[\Readable]]</a> slot is <code>false</code>,
+               <a>throw</a> an <code>InvalidStateError</code>, then abort these steps.</li>
+               <li>Let <var>data</var> be the first argument.</li>
+               <li>Transfer data from the read buffer into <var>data</var>.</li>
+               <li>Decrease the value of <var>stream</var>'s <a>[[\ReadBufferedAmount]]</a>
+               slot by the length of <var>data</var> in bytes.</li>
+               <li>Return the length of <var>data</var> in bytes.</li>
+              </ol>
```

Comment:
If there's nothing to read, do we return 0 or given an error?

Reply:
There are several situations that we need to distinguish: 

1. The stream is still open for reading, but there is nothing to read. In this case, I'd assume that readInto returns 0. 
2. The stream is no longer open for reading. In this case, throw an OperationError. 
3. The remote side called finished, and has no more to send, and everything has been read out of the read buffer (EOF).  There will be a statechange event (e.g. to "closing").  Can EOF be assumed to have been encountered after calling readInto after the transition to "closing"?